libsql/src/hrana/mod.rs

```diff
@@ -37,57 +34,36 @@ struct Cookie {
 
 /// Generic HTTP client. Needs a helper function that actually sends
 /// the request.
-#[derive(Clone, Debug)]
-pub struct Client {
-    inner: InnerClient,
-    cookies: Arc<RwLock<HashMap<u64, Cookie>>>,
-    url_for_queries: String,
-    auth: String,
-    affected_row_count: Arc<AtomicU64>,
-    last_insert_rowid: Arc<AtomicI64>,
-}
+#[derive(Debug)]
+pub struct HranaClient<T>(Arc<InnerClient<T>>);
 
-#[derive(Clone, Debug)]
-struct InnerClient {
-    inner: hyper::Client<HttpsConnector<ConnectorService>, hyper::Body>,
+impl<T> Clone for HranaClient<T> {
+    fn clone(&self) -> Self {
+        HranaClient(self.0.clone())
+    }
 }
 
-impl InnerClient {
-    fn new(connector: ConnectorService) -> Self {
-        let https = HttpsConnectorBuilder::new()
-            .with_native_roots()
-            .https_or_http()
-            .enable_http1()
-            .wrap_connector(connector);
-        let inner = hyper::Client::builder().build(https);
+impl<T> Deref for HranaClient<T> {
+    type Target = InnerClient<T>;
 
-        Self { inner }
+    fn deref(&self) -> &Self::Target {
+        self.0.deref()
     }
+}
 
-    async fn send(&self, url: String, auth: String, body: String) -> Result<ServerMsg> {
-        let req = hyper::Request::post(url)
-            .header(AUTHORIZATION, auth)
-            .body(hyper::Body::from(body))
-            .unwrap();
-
-        let res = self.inner.request(req).await.map_err(HranaError::from)?;
-
-        if res.status() != StatusCode::OK {
-            let body = hyper::body::to_bytes(res.into_body())
-                .await
-                .map_err(HranaError::from)?;
-            let body = String::from_utf8(body.into()).unwrap();
-            return Err(HranaError::Api(body));
-        }
-
-        let body = hyper::body::to_bytes(res.into_body())
-            .await
-            .map_err(HranaError::from)?;
-
-        let msg = serde_json::from_slice::<ServerMsg>(&body[..]).map_err(HranaError::from)?;
+#[derive(Debug)]
+pub struct InnerClient<T> {
```

Comment:
This code feels similar to the code in `libsql-remote`, how can we remove the duplicated code?